dietpi/dietpi-software

```diff
@@ -12318,6 +12324,40 @@ _EOF_
 			Download_Install "$(curl -sSfL 'https://api.github.com/repos/rclone/rclone/releases/latest' | mawk -F\" "/\"browser_download_url\": .*\/rclone-v[^\"\/]*linux-$arch.deb\"/{print \$4}")"
 
 		fi
+		
+		software_id=206 # Homer
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+		
+		Banner_Installing
+		
+		# installing nginx and ensuring it starts at boot
+		sudo apt install nginx -y
+		sudo systemctl enable nginx
+		
+		# installing homer
+		wget "https://github.com/bastienwirtz/homer/releases/latest/download/homer.zip" 
+		sudo unzip homer.zip -d /mnt/dietpi_userdata/Homer/
+		sudo cp /usr/share/nginx/homer/assets/config.yml.dist /mnt/dietpi_userdata/Homer/assets/config.yml
+		
+		#create nginx config - echo line by line to conf
+		sudo echo "server {" >> /etc/nginx/conf.d/homer.conf
+		sudo echo "    listen 49157;" >> /etc/nginx/conf.d/homer.conf
+		sudo echo "    listen [::]:49157;" >> /etc/nginx/conf.d/homer.conf
+		sudo echo "    root /mnt/dietpi_userdata/Homer/;" >> /etc/nginx/conf.d/homer.conf
+		sudo echo "   access_log /var/log/nginx/homer.access;" >> /etc/nginx/conf.d/homer.conf
+		sudo echo "   error_log /var/log/nginx/homer.error;" >> /etc/nginx/conf.d/homer.conf
+		sudo echo "}" >> /etc/nginx/conf.d/homer.conf
```

Comment:
configuration file can be created at once. No need to do it line by line.

Reply:
Also, is there a specific reason to not use the default Nginx config and access homer at port 80/443, as well as using default error and access log settings (access log should be generally disabled by default)? It could be made available at `/homer` sub directory.

While we usually do not do it, in such case of a jump page, and as I checked the archives content, we may think about placing it into `/var/www` directly, so not into a sub directory. It makes the uninstall a little more difficult as we need to removed every file/dir in the archives root individually, but should be okay?

Reply:
> configuration file can be created at once. No need to do it line by line.

how could this be done?

> Also, is there a specific reason to not use the default Nginx config and access homer at port 80/443

it could already be used by other applications

Reply:
> it could already be used by other applications

All webserver applications installed by `dietpi-software` are installed into their own sub directory, so they can be served on the same regular ports 80/443. I suggest then to do it the same way with Homer, so no extra config is required at all. As said, for such jump page I'd be even okay to install it into `/var/www` webroot.

Reply:
I don't recall a web server app that we have running from inside root directory. However, there could be a user own app using root dir already. Maybe better to move into an own sub folder.

Reply:
> how could this be done? 

```
     cat << '_EOF_' > /path/to/file
your
content
to
be
placed
here
_EOF_
```

Reply:
> I don't recall a web server app that we have running from inside root directory. However, there could be a user own app using root dir already. Maybe better to move into an own sub folder.

Jep, would be new to such kind of jump pages, which are only to access other web applications from. At least such kind of front pages more than any other make sense to have at webroot. But probably you're right to better be safe. Such static page is very easy to manually move to webroot, if wanted. Adding `/homer` to URL is the same additional characters than `:49157` 🙂.

Reply:
yes, if needed users can move to web server root. However, we will install following updates to /homer sub folder always. 😉